### PR TITLE
[image-manipulator][ios] Fix rotation on iOS may cause a border around the image

### LINKED ISSUE #7325: ImageManipulator.manipulateAsync with rotate option creates white border around the image
### Environment

Bare expo with following versions;
expo-image-manipulator 8.1.0
react-native-unimodules 0.8.0

Occurs on ios 13.3.1, Android 9 seems fine.

### Steps to Reproduce

When I rotate an image with manipulateAsync(), it creates a white border around the image.

Here is my code snippet

`        const rotateResult = await ImageManipulator.manipulateAsync(
            this.state.photoPreviewImageUri,
            [{ rotate: 90 }],
            { compress: 1, format: ImageManipulator.SaveFormat.JPEG }
        );`

### Expected Behavior

Rotated image without any white border

### PR DESCRIPTION
**Pull request by mnightingale**
# Why

Closes #7325 

# How

Root cause identified by @akempe >18 months ago.

A `1200 x 2000` image rotated `90 degrees` produces a `2000 x 1201` image (then after a few more rotations `2000 x 1202`).

`rotatedView.frame.size` is what is currently used which gets rounded up, then the image wont cover it leaving undrawn areas that'll be transparent for PNGs or white for JPEGs.
`rotatedSize` is with the `rounded(.down)` - probably could have used `Int(...width)` or `floor(...width)` but decided the rounded method is clearer about what is going on.

```console
po rotatedView.frame.size
▿ (2000.0, 1200.0000000000002)
  - width : 2000.0
  - height : 1200.0000000000002

po rotatedSize
▿ (2000.0, 1200.0)
  - width : 2000.0
  - height : 1200.0
```

# Test Plan

NCL and checking output image size

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image-manipulator/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 ### 🐛 Bug fixes
 
 - Remove `data:image` part in web base64 result. ([#16191](https://github.com/expo/expo/pull/16191) by [@AllanChain](https://github.com/AllanChain))
+- Fix rotation on iOS may cause a border around the image. ([#16669](https://github.com/expo/expo/pull/16669) by [@mnightingale](https://github.com/mnightingale))
```
Comment (bbarthec) [inline packages/expo-image-manipulator/CHANGELOG.md:14]:
```suggestion
- On iOS fix rotation causing extra image borders. ([#16669](https://github.com/expo/expo/pull/16669) by [@mnightingale](https://github.com/mnightingale))
```

**Review by bbarthec — APPROVED** [review]
Looking awesome 👏 
Thank you for taking care of it 😁

**Comment by antidiestro** [comment]
hey @mnightingale, thanks for this fix. We have discovered that the same issue is present when resizing an image. However I lack enough knowledge of Swift to propose a fix. I know it's a lot to ask, but is there any chance you could take a look into it? It would be a life saver. Thank you

**Comment by mnightingale** [comment]
@antidiestro please could you report a new issue ideally with a reproducable example, feel free to tag me in it and I may take a look.

After a fresh look at this issue I'm not convinced it should have rounded down rather than just rounded. I do think it's likely a similar issue affecting resizing though, the bare-expo test image is 800x533, there is a resize which results in 250x166.5625 which outputs a 250x167 image (which agrees with Photoshop resizing) but I'd hypothesize that the 166.5625 could cause issues since you can't resize to a fraction of a pixel so it might treat it as 250x166 leaving a unfilled 1px row which is very similar to what happened with rotation.

**Comment by antidiestro** [comment]
Thank you @mnightingale. Will post a repro as soon as I can.

**Pull request merged**